File: Data Structures & Algorithms/is-anagram/submission-1.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        d = dict()
        for i in s:
            if i in d.keys():
                d[i]+=1
            else:
                d[i] = 1
        for i in t:
            if i in d.keys():
                d[i]-=1
            else:
                d[i] = 1
        for i in d.keys():
            if d[i] > 0:
                return False
        return True

File: Data Structures & Algorithms/string-encode-and-decode/submission-1.cpp
class Solution {
public:
    string encode(vector<string>& strs) {
        string s;
        for(auto it: strs){
            s += it;
            s += "-";
        }
        return s;
    }

    vector<string> decode(string s) {
        vector<string> res;
        std::string temp;
        for(int i=0;i<s.size();i++){
            if(s[i] == '-'){
                string ans = temp;
                res.push_back(ans);
                temp = "";
                continue;
            } 
            temp += s[i];
        }
        return res;
    }
};
